faillint/faillint.go

```diff
@@ -201,3 +211,50 @@ func isTopName(n ast.Expr, name string) bool {
 	id, ok := n.(*ast.Ident)
 	return ok && id.Name == name && id.Obj == nil
 }
+
+func parseDirective(s string) (option string) {
+	if !strings.HasPrefix(s, "//faillint:") {
+		return ""
+	}
+	s = strings.TrimPrefix(s, "//faillint:")
+	fields := strings.Split(s, " ")
+	return fields[0]
+}
```

Comment:
I think we should check for the reason as well. And reason should be MUST have. Most of the time people add these lines and later if someone reads it they don't have an idea why it's added. Hence we should make sure it's required. I think `staticcheck` does also require reason.

Reply:
Checked now; in c11afeb3fae602236536b99b250bba01563a2f00.

Reply:
I also added a check for out-of-place `file-ignore` options; in 01ca523509364912859cd59776bebcfefa8a6b35.

Reply:
In both cases adding a test was not possible in the current framework because adding a `// want` comment negates the test conditions.  I wanted to check with you before trying anything else.  Do you want to create a `ResultType` and return something from `run()`?  Maybe a struct with a slice of `analysis.Diagnostic`?  If so do you want all the existing tests converted to use that?  Of course, any other ideas?  Maybe I'm overlooking something simple?

Manual testing,

```
$ go run . -paths errors ./faillint/testdata/src/n 
/Users/nnutter/src/github.com/nnutter/faillint/faillint/testdata/src/n/n.go:4:2: missing reason on faillint directive
exit status 3
                                                                                                                                                      (master)faillint 17:44:59 1
[1181] github.com/nnutter/faillint $ go run . -paths errors ./faillint/testdata/src/o
/Users/nnutter/src/github.com/nnutter/faillint/faillint/testdata/src/o/o.go:4:2: file-ignore option on faillint directive must be in package docs
/Users/nnutter/src/github.com/nnutter/faillint/faillint/testdata/src/o/o.go:5:2: package "errors" shouldn't be imported
exit status 3
```

Reply:
I think we should have a `TestParseDirective` that tests the `parseDirective` function. And then you can check for various parse directives and I think that would be sufficient for our needs.

Reply:
I merged our two ideas together. =)

Reply:
Thanks, this is much better 👍